dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/St6c.mo

```diff
@@ -0,0 +1,351 @@
+  Modelica.Blocks.Sources.Constant const1(k = IlrPu * Kcl) annotation(
+    Placement(visible = true, transformation(origin = {-330, 20}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Gain gain(k = Klr) annotation(
+    Placement(visible = true, transformation(origin = {-170, 20}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Max max3 annotation(
+    Placement(visible = true, transformation(origin = {-110, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const2(k = VrMinPu) annotation(
+    Placement(visible = true, transformation(origin = {-170, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Min min3 annotation(
+    Placement(visible = true, transformation(origin = {-50, 20}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const3(k = VaMaxPu) annotation(
+    Placement(visible = true, transformation(origin = {-110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const4(k = VaMinPu) annotation(
+    Placement(visible = true, transformation(origin = {-110, -200}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Sum sum1(nin = 4) annotation(
+    Placement(visible = true, transformation(origin = {-290, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.Sum sum2(nin = 5) annotation(
+    Placement(visible = true, transformation(origin = {-110, -100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.MinMax max1(nu = 3) annotation(
+    Placement(visible = true, transformation(origin = {-230, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.MinMax min1(nu = 3) annotation(
+    Placement(visible = true, transformation(origin = {-170, -54}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.MinMax max2(nu = 3) annotation(
+    Placement(visible = true, transformation(origin = {150, -80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Math.MinMax min2(nu = 3) annotation(
+    Placement(visible = true, transformation(origin = {210, -74}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Logical.Switch switch annotation(
+    Placement(visible = true, transformation(origin = {-330, 100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.Constant const5(k = Kp) annotation(
+    Placement(visible = true, transformation(origin = {-390, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Sources.BooleanConstant booleanConstant(k = Sw1) annotation(
+    Placement(visible = true, transformation(origin = {-390, 100}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.FirstOrder firstOrder2(T = tLimMax, y_start = IlrPu * Kcl - Ir0Pu) annotation(
```

Comment:
It avoids a simulation failure. I am trying to get rid of this block.

Reply:
The FirstOrder block has been deleted. In order to ensure the success of the simulations relying on St6c (PmConstSt6bPss3b, PmConstSt6cPss6c) :
- the minimalAcceptableStep has been set to 1e-8
- the PIAntiWindupVariableLimits block now uses the VariableLimiter from Dynawo (thus omitting the assert causing a simulation failure if both limits are equal)
- for PmConstSt6bPss3b, tA has been set to 1e-4 since 1e-5 leads to a simulation failure